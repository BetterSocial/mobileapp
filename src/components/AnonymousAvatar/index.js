import * as React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';

import AnonymousProfile from '../../assets/images/AnonymousProfile.png';
import dimen from '../../utils/dimen';
import {POST_VERSION} from '../../utils/constants';

export const styles = StyleSheet.create({
  imageAnonimity: {
    marginRight: 0,
    width: dimen.size.FEED_HEADER_IMAGE_RADIUS,
    height: dimen.size.FEED_HEADER_IMAGE_RADIUS
  },
  avatarV2Background: (backgroundColor, radius) => ({
    width: radius,
    height: radius,
    borderRadius: radius / 2,
    backgroundColor,
    justifyContent: 'center'
  }),
  avatarV2Emoji: (emojiRadius) => ({
    fontSize: emojiRadius,
    textAlign: 'center',
    alignSelf: 'center'
  })
});

/**
 * @typedef {Object} AnonymousAvatarComponentProps
 * @property {number} [radius=dimen.size.FEED_HEADER_IMAGE_RADIUS]
 * @property {number} [emojiRadius=dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS]
 * @property {string} version
 * @property {import('react-native').StyleProp} containerStyle
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousAvatarComponentProps} props
 */
const AnonymousAvatar = (props) => {
  const {
    version,
    anonUserInfo,
    containerStyle = {},
    radius = dimen.size.FEED_HEADER_IMAGE_RADIUS,
    emojiRadius = dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS
  } = props;

  if (version >= POST_VERSION) {
    return (
      <View
        testID="newVersion"
        style={{...styles.avatarV2Background(anonUserInfo.colorCode, radius), ...containerStyle}}>
        <Text style={styles.avatarV2Emoji(emojiRadius)}>{anonUserInfo.emojiCode}</Text>
      </View>
    );
  }

  return (
    <Image
      source={AnonymousProfile}
      width={radius}
      height={radius}
      style={{...styles.imageAnonimity, ...containerStyle}}
    />
  );
};

export default AnonymousAvatar;
