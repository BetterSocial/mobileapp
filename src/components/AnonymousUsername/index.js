import * as React from 'react';
import {StyleSheet, Text} from 'react-native';

import {POST_VERSION} from '../../utils/constants';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const styles = StyleSheet.create({
  feedUsername: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 14,
    lineHeight: 16.94,
    color: colors.black,
    flex: 1
  }
});

/**
 * @typedef {Object} AnonymousUsernameComponentProps
 * @property {string} version
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousUsernameComponentProps} props
 */
const AnonymousUsername = (props) => {
  const {version, anonUserInfo} = props;

  if (version >= POST_VERSION) {
    return (
      <Text
        testID="newVersion"
        style={styles.feedUsername}>{`${anonUserInfo.colorName} ${anonUserInfo.emojiName}`}</Text>
    );
  }

  return <Text style={styles.feedUsername}>Anonymous</Text>;
};

export default AnonymousUsername;
