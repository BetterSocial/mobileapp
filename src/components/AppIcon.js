import React from 'react';
import {Image, StyleSheet} from 'react-native';

function AppIcon() {
  return (
    <Image
      source={require('./../assets/ping.png')}
      style={styles.logo}
      resizeMode="cover"
    />
  );
}

const styles = StyleSheet.create({
  logo: {
    width: 41,
    height: 52,
  },
});

export default AppIcon;
