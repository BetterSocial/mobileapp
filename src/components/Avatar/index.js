import * as React from 'react';
import {View, StyleSheet, Image} from 'react-native';

const index = ({image, style}) => {
  return (
    <View style={[styles.wrapperImage, style]}>
      <Image
        source={{
          uri: image
            ? image
            : 'https://res.cloudinary.com/hpjivutj2/image/upload/v1617245336/Frame_66_1_xgvszh.png',
        }}
        style={[styles.image, StyleSheet.absoluteFillObject]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  wrapperImage: {
    borderRadius: 45,
    borderWidth: 0.2,
    borderColor: 'rgba(0,0,0,0.5)',
    width: 46,
    height: 46,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: '100%',
    width: '100%',
    borderRadius: 45,
  },
});

export default index;
