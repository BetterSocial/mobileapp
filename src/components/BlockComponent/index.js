/* eslint-disable no-underscore-dangle */
import * as React from 'react';
import Toast from 'react-native-simple-toast';
import { InteractionManager, View } from 'react-native';

import BlockPostAnonymous from '../Blocking/BlockPostAnonymous';
import BlockUser from '../Blocking/BlockUser';
import ReportPostAnonymous from '../Blocking/ReportPostAnonymous';
import ReportUser from '../Blocking/ReportUser';
import SpecificIssue from '../Blocking/SpecificIssue';
import blockUtils from '../../service/utils/blockUtils';
import { getUserId } from '../../utils/users';

class BlockComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            messageReport: '',
            myId: '',
            postId: '',
            reportOption: -1,
            userId: '',
            username: '',
            reason: [],
            isAnonymous: false,
        };

        this.refBlockPostAnonymous = React.createRef();
        this.refBlockUser = React.createRef();
        this.refReportPostAnonymous = React.createRef();
        this.refReportUser = React.createRef();
        this.refSpecificIssue = React.createRef();

        this.__blockPostAnonymous = this.__blockPostAnonymous.bind(this)
        this.__blockUser = this.__blockUser.bind(this)
        this.__onIssue = this.__onIssue.bind(this)
        this.__onNextQuestion = this.__onNextQuestion.bind(this);
        this.__onSelectBlocking = this.__onSelectBlocking.bind(this);
        this.__onSelectBlockingPostAnonymous = this.__onSelectBlockingPostAnonymous.bind(this)
        this.__onSkipOnlyBlock = this.__onSkipOnlyBlock.bind(this)
    }

    componentDidMount() {
        this.__parseToken()
    }


    setDataToState(value) {
        if (value.anonimity === true) {
            this.setState({
                postId: value.id,
                userId: `${value.actor.id}-anonymous`,
                username: 'Anonymous',
            })
        } else {
            this.setState({
                postId: value.id,
                userId: value.actor.id,
                username: value.actor.data.username,
            })
        }
    };

    openBlockComponent(value) {
        if (value.actor.id === this.state.myId) {
            Toast.show("Can't Block yourself", Toast.LONG);
        } else {
            this.setDataToState(value);
            if (value.anonimity) {
                this.setState({ isAnonymous: true })
                this.refBlockPostAnonymous.current.open();
            } else {
                this.setState({ isAnonymous: false })
                this.refBlockUser.current.open();
            }
        }
    }

    openReportAnonymousPost() {
        this.refBlockPostAnonymous().current.open()
    }

    async __parseToken() {
        const id = await getUserId();
        if (id) {
            this.setState({
                myId: id
            })
        }
    };

    __onSelectBlocking(v) {
        if (v !== 1) {
            InteractionManager.runAfterInteractions(() => {
                this.refReportUser.current.open()
            })
        } else {
            this.__blockUser();
        }
        this.refBlockUser.current.close();
    }

    __onSelectBlockingPostAnonymous(v) {
        if (v !== 1) {
            InteractionManager.runAfterInteractions(() => {
                this.refReportPostAnonymous.current.open();
            })
        } else {
            this.__blockPostAnonymous();
        }
        this.refBlockPostAnonymous.current.close();
    }

    __onNextQuestion(v) {
        this.setState({
            reason: v
        });
        this.refReportUser.current.close();
        this.refReportPostAnonymous.current.close();
        InteractionManager.runAfterInteractions(() => {
            this.refSpecificIssue.current.open();
        })
    }

    __onSkipOnlyBlock() {
        this.refReportUser.current.close();
        if (this.state.isAnonymous) return this.__blockPostAnonymous();
        return this.__blockUser();
    }

    __onIssue(v) {
        this.refSpecificIssue.current.close();

        this.setState({
            messageReport: v
        })

        setTimeout(() => {
            this.refBlockUser.current.close()
            this.refBlockPostAnonymous.current.close()

            if (this.state.isAnonymous) return this.__blockPostAnonymous();
            return this.__blockUser();
        }, 500);
    }

    __blockUser() {
        if (__DEV__) {
            console.log('block user called');
        }
        const { postId, userId, messageReport, reason } = this.state
        blockUtils.uiBlockUser(
            postId,
            userId,
            this.props.screen || "screen_feed",
            reason,
            messageReport,
            () => {
                if(this.props.refresh) {
                    this.props.refresh(this.state.postId)
                }
            }
        )
    }

    __blockPostAnonymous() {
        const { postId, messageReport, reason } = this.state
        blockUtils.uiBlockPostAnonymous(
            postId,
            this.props.screen || "screen_feed",
            reason,
            messageReport,
            () => {
                if(this.props.refreshAnonymous) {
                     this.props.refreshAnonymous(this.state.postId)
                }
            }
        )
    }

    render() {
        return <View>
            <BlockPostAnonymous
                refBlockPostAnonymous={this.refBlockPostAnonymous}
                onSelect={this.__onSelectBlockingPostAnonymous}
            />

            <BlockUser
                refBlockUser={this.refBlockUser}
                onSelect={this.__onSelectBlocking}
                username={this.state.username}
            />
            <ReportUser
                ref={this.refReportUser}
                onSelect={this.__onNextQuestion}
                onSkip={this.__onSkipOnlyBlock}
            />
            <ReportPostAnonymous
                refReportPostAnonymous={this.refReportPostAnonymous}
                onSelect={this.__onNextQuestion}
                onSkip={this.__onSkipOnlyBlock}
            />
            <SpecificIssue
                refSpecificIssue={this.refSpecificIssue}
                onPress={this.__onIssue}
                onSkip={this.__onSkipOnlyBlock}
            />
        </View>
    }
}

export default BlockComponent