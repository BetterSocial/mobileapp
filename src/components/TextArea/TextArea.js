import * as React from 'react';
import {StyleSheet, TextInput} from 'react-native';

import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const TextArea = ({
  textAlignVertical = 'top',
  multiline = true,
  placeholder = '',
  onChangeText = () => {},
  value = '',
  style = {},
  onRef = (ref) => {}
}) => {
  const textRef = React.useRef();
  React.useEffect(() => {
    onRef(textRef);
  }, []);

  return (
    <TextInput
      ref={textRef}
      autoFocus={true}
      value={value}
      onChangeText={onChangeText}
      multiline={multiline}
      style={{...styles.input, ...style}}
      textAlignVertical={textAlignVertical}
      placeholder={placeholder}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    backgroundColor: COLORS.lightgrey,
    paddingVertical: 16,
    paddingHorizontal: 12,
    height: 150,
    justifyContent: 'flex-start',
    overflow: 'scroll',
    borderRadius: 8,
    fontFamily: fonts.inter[500],
    fontSize: 14,
    color: COLORS.black,
    lineHeight: 24
  }
});
export default TextArea;
