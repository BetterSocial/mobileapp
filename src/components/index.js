import Avatar from './Avatar';
import SingleSidedShadowBox from './SingleSidedShadowBox';
import Footer from './Footer/Footer';

export {Avatar, SingleSidedShadowBox, Footer};
