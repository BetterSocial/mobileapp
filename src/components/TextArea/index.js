import TextArea from './TextArea';
import AutoFocusTextArea from './AutoFocusTextArea';
export {TextArea, AutoFocusTextArea};
