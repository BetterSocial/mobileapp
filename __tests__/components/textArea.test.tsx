import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {TextArea, AutoFocusTextArea} from '../../src/components/TextArea';

configure({adapter: new Adapter()});
describe('component TextArea', () => {
  it('TextArea snapshot', () => {
    const component = shallow(
      <TextArea
        textAlignVertical="top"
        multiline={true}
        placeholder="Input Text"
        onChangeText={() => {}}
        value="Input"
        style={{}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('TextArea renders correctly', () => {
    const tree = renderer
      .create(
        <TextArea
          textAlignVertical="top"
          multiline={true}
          placeholder="Input Text"
          onChangeText={() => {}}
          value="Input"
          style={{}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AutoFocusTextArea snapshot', () => {
    const component = shallow(
      <AutoFocusTextArea
        placeholder="Input Text"
        onChangeText={() => {}}
        value="Input"
        style={{}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('AutoFocusTextArea renders correctly', () => {
    const tree = renderer
      .create(
        <AutoFocusTextArea
          placeholder="Input Text"
          onChangeText={() => {}}
          value="Input"
          style={{}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
