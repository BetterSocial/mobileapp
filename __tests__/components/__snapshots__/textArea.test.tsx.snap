// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component TextArea AutoFocusTextArea renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  multiline={true}
  onChangeText={[Function]}
  placeholder="Input Text"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "borderRadius": 8,
      "color": "#000000",
      "fontFamily": "Inter-Medium",
      "fontSize": 14,
      "height": 150,
      "justifyContent": "flex-start",
      "lineHeight": 24,
      "overflow": "scroll",
      "paddingHorizontal": 12,
      "paddingVertical": 16,
    }
  }
  textAlignVertical="top"
  underlineColorAndroid="transparent"
  value="Input"
/>
`;

exports[`component TextArea AutoFocusTextArea snapshot 1`] = `
<Component
  allowFontScaling={true}
  multiline={true}
  onChangeText={[Function]}
  placeholder="Input Text"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "borderRadius": 8,
      "color": "#000000",
      "fontFamily": "Inter-Medium",
      "fontSize": 14,
      "height": 150,
      "justifyContent": "flex-start",
      "lineHeight": 24,
      "overflow": "scroll",
      "paddingHorizontal": 12,
      "paddingVertical": 16,
    }
  }
  textAlignVertical="top"
  underlineColorAndroid="transparent"
  value="Input"
/>
`;

exports[`component TextArea TextArea renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  autoFocus={true}
  multiline={true}
  onChangeText={[Function]}
  placeholder="Input Text"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "borderRadius": 8,
      "color": "#000000",
      "fontFamily": "Inter-Medium",
      "fontSize": 14,
      "height": 150,
      "justifyContent": "flex-start",
      "lineHeight": 24,
      "overflow": "scroll",
      "paddingHorizontal": 12,
      "paddingVertical": 16,
    }
  }
  textAlignVertical="top"
  underlineColorAndroid="transparent"
  value="Input"
/>
`;

exports[`component TextArea TextArea snapshot 1`] = `
<Component
  allowFontScaling={true}
  autoFocus={true}
  multiline={true}
  onChangeText={[Function]}
  placeholder="Input Text"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "borderRadius": 8,
      "color": "#000000",
      "fontFamily": "Inter-Medium",
      "fontSize": 14,
      "height": 150,
      "justifyContent": "flex-start",
      "lineHeight": 24,
      "overflow": "scroll",
      "paddingHorizontal": 12,
      "paddingVertical": 16,
    }
  }
  textAlignVertical="top"
  underlineColorAndroid="transparent"
  value="Input"
/>
`;
